### ISSUE TITLE
Music export for CTR, Bash, and possibly other SEQ-using games

### ISSUE THREAD

**Issue by NetyashaRoozi** [cbhacks/CrashEdit#96]
I know its not a issue but I suck at github and I wanted to request this, I know crash bash and crash team racing have the same type of music, Now I know that crash bash probably wont be supported but I would be super happy if it was possible to mod Crash Team Racing and to rip the levels/models

**Comment by ManDude** [comment]
There is a CTR ripper and Bash's music is... weird.

**Comment by NetyashaRoozi** [comment]
Wait..... there is? but I love the music in crash bash, but some of them don't fit like the music for polar panic.... so weird but it has a cool tune to it

**Comment by NetyashaRoozi** [comment]
I cant find the CTR ripper anywhere I just get ripper roo and other junk, By any chance do you have a link?

**Comment by ughman** [comment]
Is there any evidence that CTR actually uses SEQ or MIDI? The readme for the PSF rip I saw once claimed it used some custom naughtydog music engine named HOWL (not an exact quote).

I do know that Bash uses SEQ though, I think nextvolume ripped them all out once.

**Comment by NetyashaRoozi** [comment]
Well I have explored the file and I know for a fact its midi-like, CTR uses a format called Minipsf which contains sequences and a Instrument Container (Soundfont), I have not explored Crash Bash yet but I have ripped the instruments from it so I know it also has a Midi-like format

**Comment by ughman** [comment]
> CTR uses a format called Minipsf

No offense, but that doesn't mean anything. Except for the ones that use CD-DA or CD-XA, every playstation game "uses" PSF because PSF is literally just playstation executable code. PSF playback plugins are playstation emulators. It's like saying your Win32 game uses EXE format music.

**Comment by AraHaan** [comment]
FSoundfonts are good are the sound fonts of like swf or soemthign like that file type? If you want you can send the files over to me sometime through a github repo link and i can download them to see if the XMPlay Plgin can play them as it needs and loads up soundfonts for MIDI's So maybe a simple rename of the files to midi could make them playable in XMPlay. Also tit is possibkle to have exe format it is called a xm-exe you can look at it here http://www.un4seen.com/xmexe.htm
And coem to think about it it is possible to use Bass.net with Bass to play a wav file as you can do all sorts of stuff with it liek a reverb and stuff its awesome. So music wise you can always export a temp wav file for bass to play and when done have it be cleaned up.

**Comment by ManDude** [comment]
CrashEdit can export to DLS. You can then convert DLS to other formats (make sure you manually fix the DLS though).

**Comment by AraHaan** [comment]
true, but what Crash Edit cant do is convert the VAB files to a soundfont for use in XMPlay where there you can use the WAV writer on that to combine them together as if you are in game.

**Comment by NetyashaRoozi** [comment]
Just a tiny update, I ripped all most all of the music from Crash Bash and exported them into midi's and soundfonts, 2 songs have all the tracks combined and those tracks are the Loading theme and the unused loading theme. Some tracks don't rip correctly and can only rip the sequence and not the sounds, these tracks contain multiple sample collections, for example Crash Ball has 2 sample collections and attempting to rip them will cause most of the sounds to distort. I am working on trying to fix this issue. I found the soundtrack to CTR in psf form, It plays perfectly in Winamp using the PSF plugin but if you attempt to rip anything from it you will only get the samples and most of the samples will be distorted or silent. This is because CTR uses a custom psf format (You guys already know this by now) I will still be tinkering with Crash Team Racing because I really want to rip the samples into a soundfont like file and I have always wanted to customize the CTR music and tracks

**Comment by ManDude** [comment]
I already ripped all the music from Crash Bash, don't bother.

**Comment by NetyashaRoozi** [comment]
Does anyone know a way to rip Crash Team Racing music into sequences and samples yet? I still can't find a way to extract sequences and the only way I could extract samples was through PSound

**Comment by ManDude** [comment]
Nope. It uses some weird custom sequence format and nobody has bothered to reverse-engineer it yet.

**Comment by Sylvielmna** [comment]
Hey, this is old, but I bothered to reverse engineer it a while ago. Check out the format pdf.
https://github.com/Sylvielmna/cseq2midi
